### ISSUE TITLE
Confused by behaviour of addition

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaDiff/TaylorSeries.jl#225]
```julia
julia> Taylor1(1)
 1.0 t + 𝒪(t²)

julia> Taylor1(Float64,10)^100
 0.0 + 𝒪(t¹¹)

julia> Taylor1(1) + (Taylor1(Float64,10)^100)
 1.0 t + 𝒪(t¹¹)
```

I would have expected `(t + O(t^2)) + O(t^11) == t + O(t^2)`.

**Comment by dlfivefifty** [comment]
More broadly, I'm curious about 
```julia
julia> Taylor1(1)
 1.0 t + 𝒪(t²)
```
I would have thought it would be `1.0t + O(t^∞)`.

**Comment by lbenet** [comment]
Thanks for asking. Let me begin saying that this is the expected/intended result "by design". The design decision at some point was that the result for operations involving two polynomials of different order should be a `Taylor1` polynomial with the maximum order. In the first example you quote, this is precisely what it happens.

I guess that the origin of the confusion is the fact that the output has a big 𝒪. We use it simply to know the order of the "truncated" polynomial, rather than in the mathematical sense. I thought that this was a "neat" way of expressing the fact that the polynomials are truncated, without any further implication.

Do you have a suggestion to improve this? Documenting this would suffice?

**Comment by dlfivefifty** [comment]
I guess why not the minimal order? As it is now it feels mathematically ill-defined, unless I'm missing something.

Note we are facing the same issues in https://github.com/idlemat/PowerNumbers.jl

**Comment by lbenet** [comment]
I agree with you that mathematically it is strange and perhaps ill-defined. That's the reason that maybe changing the output would be the best: perhaps we should output `Taylor1(10)` as `1.0 t + 𝒯(t¹¹)`, the 𝒯 meaning "truncated at order 10".

**Comment by lbenet** [comment]
I actually prefer to "promote" things to the largest order...

**Comment by lbenet** [comment]
I agree that @dlfivefifty has a point. 

I have implemented the changes in #226. Locally, tests pass, though I think more tests may be needed. The changes in the docs are still to be implemented.

@dlfivefifty Can you take a look if things now are more consistent?
